core/src/utilities/random.cc

```diff
@@ -382,6 +382,94 @@ namespace jiminy
         return lerp(ratio, yLeft, yRight);
     }
 
+    double AbstractPerlinNoiseOctave::gradient(double t) const
+    {
+        // Get current phase
+        const double phase = t / wavelength_ + shift_;
+
+        // Compute closest right and left knots
+        const int32_t phaseIndexLeft = static_cast<int32_t>(phase);
+        const int32_t phaseIndexRight = phaseIndexLeft + 1;
+
+        // Compute smoothed ratio of current phase wrt to the closest knots
+        const double dtLeft = phase - phaseIndexLeft;
+        const double dtRight = dtLeft - 1.0;
+        const double ratio = fade(dtLeft);
+        const double dRatio = fade_derivative(dtLeft);
+
+        /* Compute gradients at knots, and perform linear interpolation between them to get value
+           at current phase.*/
+        const double yLeft = grad(phaseIndexLeft, dtLeft);
+        const double yRight = grad(phaseIndexRight, dtRight);
+        return (yRight - yLeft) * dRatio + lerp(ratio, yLeft / dtLeft, yRight / dtRight);
+    }
+
+    double AbstractPerlinNoiseOctave::operator()(double x, double y) const
```

Comment:
I don't understand why do you need all. 2D Perlin process is not the same as the sum of 1D processes along x-axis and y-axis respectively ? If so, just do this.

Reply:
<img width="1459" alt="random_perlin_ground_bidirectional" src="https://github.com/duburcqa/jiminy/assets/101942083/e2131df6-e386-4e19-a007-0373aa0bf4a5">

2D Perlin process is not the same as the sum of 1D processes along x-axis and y-axis. Here's a photo of the sum (to compare with the photo in the description). That's what I did first and then realize quickly it wasn't what I was looking for.

Reply:
Ok now I see the issue. It is indeed more tricky than exacted. I'm going to think about it to see if there is a way to get the same result without duplicating everything. The big issue is that the contribution of each process independent on each axis. I'm wondering if passing the 2D position encoding (hash) to two independent random processes should do the trick.

Reply:
After check, it is indeed more tricky than that and you really need a 2D implementation. Position encoding is not enough per se. I forgot about random Perlin processes were evaluated in the first place. Since it is an interpolation, it is always necessary to have a dedicated 2D implementation.